core/daemon/src/main/java/org/opennms/netmgt/vmmgr/Invoker.java

```diff
@@ -137,15 +148,55 @@ public void instantiateClasses() {
                     }
                 }
             } catch (Throwable t) {
-		LOG.error("An error occurred loading the mbean {} of type {}", service.getName(), service.getClassName(), t);
+                LOG.error("An error occurred loading the mbean {} of type {}", service.getName(), service.getClassName(), t);
                 invokerService.setBadThrowable(t);
             }
         }
     }
 
-    /**
-     * <p>getObjectInstances</p>
-     */
+    private void registerSignalHandlerOnce() {
+        if (s_handler != null) return;
+
+        try {
+            s_handler = Signal.handle(new Signal("USR1"), signal -> writeStatusUpdate());
+        } catch (final Throwable t) {
+            System.err.println("WARNING: failed to register service status signal handler");
+            t.printStackTrace();
+        }
+    }
+
+    private void writeStatusUpdate() {
+        if (m_statusPath == null) {
+            System.err.println("WARNING: status signal triggered, but no status path is configured!");
+            return;
+        }
+
+        final var output = new StringBuilder();
+
+        for (final var invokerService : getServices()) {
+            final var serviceName = invokerService.getService().getName();
+            for (final var invoke : invokerService.getService().getInvokes()) {
+                if ("status".equals(invoke.getMethod())) {
+                    try {
+                        Object result = invoke(invoke, invokerService.getMbean());
+                        output.append(serviceName).append("=").append(result.toString()).append("\n");
```

Comment:
this could potentially be a cleaner place to format the output

Reply:
true; I originally avoided it to just dump the data simply, but might as well format it when it's easily parseable in the other style